### DISCUSSION TITLE
setup_password_recovery_routes & registration_open keys

### DISCUSSION THREAD

**Discussion by blondie63** [Laravel-Backpack/community-forum#673] [Q&A (Help)] [answered]
During BKP6 theme conversion to Coreuiv2 i've this warnings in laravel.log

```
[2023-09-12 14:11:14] local.ERROR: Could not find config key: setup_password_recovery_routes. Neither in the Backpack theme, nor in the fallback theme, nor in ui.  
[2023-09-12 14:11:14] local.ERROR: Could not find config key: registration_open. Neither in the Backpack theme, nor in the fallback theme, nor in ui.  
```

but i've these keys in config/backpack/base.php Same like bkp demo..

What's wrong ?

**Comment by jcastroa87** [comment]
Hello @blondie63

We are reviewing the message you sent us and we have not been able to replicate the error in a new installation, could you confirm that all Backpack packages are updated (CRUD, PRO, themes, etc.) and if this was an update, you didn't make any customizations to the settings?

Thank you

**Reply by blondie63** [reply]
Yes, sure @jorgetwgroup !
```
### PHP VERSION:
PHP 8.2.10 (cli) (built: Aug 31 2023 18:52:55) (NTS)
Copyright (c) The PHP Group
Zend Engine v4.2.10, Copyright (c) Zend Technologies
    with Xdebug v3.2.2, Copyright (c) 2002-2023, by Derick Rethans
    with Zend OPcache v8.2.10, Copyright (c), by Zend Technologies

### LARAVEL VERSION:
10.23.0.0

### BACKPACK PACKAGE VERSIONS:
backpack/basset: 1.2.0
backpack/crud: 6.2.0
backpack/devtools: 2.0.3
backpack/editable-columns: 3.0.2
backpack/generators: v4.0.2
backpack/pro: 2.0.11
backpack/settings: 3.1.0
backpack/theme-coreuiv2: 1.2.0
backpack/theme-tabler: 1.1.0
```

**Comment by tabacitu** [comment]
@blondie63 try searching for `setup_password_recovery_routes` in the entire project (including the vendor directory). That will tell you where that config is used. Can you give us a list?

**Comment by blondie63** [comment]
@tabacitu, these the two search:
 
<img width="525" alt="bkp6-set_psw_search1" src="https://github.com/Laravel-Backpack/community-forum/assets/1261851/ab5eb7d3-571e-4f50-ab92-bccdfb71803a">
<img width="583" alt="bkp6-set_psw_search2" src="https://github.com/Laravel-Backpack/community-forum/assets/1261851/525ecd75-13a0-4009-8a4f-b0ed4da99f34">

**Comment by jcastroa87** [comment]
As i see in code, thats error is show from method "backpack_theme_config", so is needed to check on your login.blade.php that call.

And did you try to clean cache? `php artisan config:clear`

**Reply by blondie63** [reply]
Hi @jorgetwgroup attached two my blades
[Login_Register_blades.zip](https://github.com/Laravel-Backpack/community-forum/files/12605301/Login_Register_blades.zip)

**Comment by jcastroa87** [comment, marked as answer]
If you see in https://github.com/Laravel-Backpack/theme-coreuiv2/blob/main/resources/views/auth/login.blade.php

the line 84 is:
```
@if (backpack_users_have_email() && backpack_email_column() == 'email' && config('backpack.base.setup_password_recovery_routes', true))
```

and yours is:
```
@if (backpack_users_have_email() && backpack_theme_config('setup_password_recovery_routes', true))
```

Need to to the same with line 89
Original
```
@if (config('backpack.base.registration_open'))
```

Yours:
```
@if (backpack_theme_config('registration_open'))
```

Try to adapt as the official file, refresh the cache and the error must gone

Cheers

**Reply by blondie63** [reply]
To make things quick I did a massive "Search & Replace" but something went wrong!

Thanks

**Comment by blondie63** [comment]
@jorgetwgroup @tabacitu Thanks for all, now only this #668 is missing to me :(

**Comment by blondie63** [comment]
I've found the problem !! Issues about bootstrap include !
